jwst/outlier_detection/outlier_detection.py

```diff
@@ -361,13 +361,10 @@ def flag_cr(sci_image, blot_image, **pars):
     snr1, snr2 = [float(val) for val in pars.get('snr', '5.0 4.0').split()]
     scl1, scl2 = [float(val) for val in pars.get('scale', '1.2 0.7').split()]
 
-    if sci_image.meta.background.subtracted:
+    if not sci_image.meta.background.subtracted:
```

Comment:
I don't understand why the logic has been flipped here. If the background has NOT already been subtracted, then why is it being reported as subtracted and the (unsubtracted) level passed along for use elsewhere?

Reply:
The background level does NOT get subtracted by skymatch from the original input _cal.fits files.  The value only gets applied when resampling the image.  Thus, when outlier_detection gets run, it needs to convert a background-subtracted-median-blotted image with the original non-background-subtracted values in the input _cal file.  If a change was made or the user performed background-subtraction on the _cal.fits file before running outlier_detection, then we would not need to get the background value to amend the background-subtracted-blot image.   I agree it seems backwards on the face of it, but I believe it insures the right values are used for the detection of bad pixels/cosmic-rays.

Reply:
👍